core/src/main/java/org/mule/runtime/core/routing/IdempotentMessageFilter.java

```diff
@@ -134,6 +128,27 @@ protected boolean accept(Event event, Event.Builder builder) {
     }
   }
 
+  @Override
+  public final Event process(Event event) throws MuleException {
+    if (accept(event)) {
+      return processNext(event);
+    } else {
+      event.getContext().success();
```

Comment:
The chain should do this in `process` code path.  I need to check if that is correctly implemented, but this shouldn't be needed here.